### PR TITLE
cocoapods: 1.10.2 -> 1.11.0

### PR DESCRIPTION
**Pull request by winterqt**
###### Motivation for this change

This updates CocoaPods to 1.11.0, and with it, [fixes a bug in the update script involving the Gemfile platform not being set to `ruby`](https://github.com/NixOS/nixpkgs/pull/134396#discussion_r703389621).

@jakubgs, mind making sure that this fixes your issue?

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/mobile/cocoapods/update [hunk 1]
```diff
@@ -5,8 +5,16 @@ set -e
 
 cd "$(dirname "${BASH_SOURCE[0]}")"
 
+PLATFORM=$(bundler platform | head -n 1 | cut -d ":" -f 2 | cut -c 2-)
 rm -f Gemfile.lock Gemfile-beta.lock
+
 bundler lock
```
Comment (marsam) [inline pkgs/development/mobile/cocoapods/update:11]: Let's force a platform-independent lock:
```suggestion
BUNDLE_FORCE_RUBY_PLATFORM=1 bundler lock
```
with this variable, the other changes to this file aren't needed

File: pkgs/development/mobile/cocoapods/update [hunk 2]
```diff
@@ -5,8 +5,16 @@ set -e
...
+bundler lock --add-platform ruby
+bundler lock --remove-platform $PLATFORM
+
 BUNDLE_GEMFILE=Gemfile-beta bundler lock --lockfile=Gemfile-beta.lock
```
Comment (marsam) [inline pkgs/development/mobile/cocoapods/update:15]: ditto
```suggestion
BUNDLE_FORCE_RUBY_PLATFORM=1 BUNDLE_GEMFILE=Gemfile-beta bundler lock --lockfile=Gemfile-beta.lock
```

**Review by marsam — CHANGES_REQUESTED** [review]

File: pkgs/development/mobile/cocoapods/update (hunk 1 above)
Comment (jakubgs) [reply pkgs/development/mobile/cocoapods/update:11]: That's neat. Much better than using the flags.

**Review by jakubgs — APPROVED** [review]
I agree with comments from @marsam .

For what it's worth I tested this change in our mobile app repo and it works: https://github.com/status-im/status-react/pull/11900

File: pkgs/development/mobile/cocoapods/update (hunk 1 above)
Comment (winterqt) [reply pkgs/development/mobile/cocoapods/update:11]: Thank you, I didn't know that environment variable existed!

**Review by marsam — APPROVED** [review]

**Pull request merged**